server.js

```diff
@@ -6,22 +6,112 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
+const listEndpoints = require('express-list-endpoints');
+
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+const { Schema } = mongoose;
+
+const ThoughtSchema = new Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  heart: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: () => Date.now()
+  }
+}); 
+
+const Thought = mongoose.model("Thought", ThoughtSchema);
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).send({
+    succes: true,
+    message: "OK",
+    body: {
+      conent: "Happy Thoughs API",
+      endpoints: listEndpoints(app)
+    }
+  });
+});
+
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find()
+    .sort({ createdAt: 'desc', _id: 'desc' })
+    .limit(20)
+    .exec();
+  res.status(200).json(thoughts);
+});
+
+
+app.get("/thoughts/id/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const singleThought = await Thought.findById(id)
+    if (singleThought) {
+      res.status(200).json(singleThought);
+    } else {
+      res.status(400).json('Not found');
+    }
+    } catch (e) {
+      res.status(400).json('No id found');
+    }
 });
 
-// Start the server
+app.post("/thoughts", async (req, res) =>{
+  const { message } = req.body;
+    try {
+      const thought = await new Thought({ message }).save();
+      res.status(201).json({
+        succes: true,
+        response: thought,
+        message: "Created with success"
+      });
+    } catch (e) {
+      res.status(400).json({
+        succes: false,
+        response: e,
+        message: "Error occured when posting"
+      });
+    }
+});
+
+app.patch("/thoughts/:_id/like", async (req, res) => {
+  const { _id } = req.params; 
+  try {
+    const thought = await Thought.findByIdAndUpdate(
+      _id,
+      { $inc: { heart: 1 } },
```

Comment:
Really good job using the mongoose `$inc` operator for your likes. I did not think of that!